src/Tickets/Seating/Orders/Attendee.php

```diff
@@ -0,0 +1,72 @@
+<?php
+/**
+ * Manage seat selection data for attendee.
+ *
+ * @since TBD
+ *
+ * @package TEC/Tickets/Seating/Orders
+ */
+
+namespace TEC\Tickets\Seating\Orders;
+
+use TEC\Tickets\Seating\Meta;
+use Tribe__Main as Common;
+use TEC\Tickets\Admin\Attendees\Page as Attendees_Page;
+
+/**
+ * Class Attendee
+ *
+ * @since TBD
+ *
+ * @package TEC/Tickets/Seating/Orders
+ */
+class Attendee {
+	/**
+	 * Adds the attendee seat column to the attendee list.
+	 *
+	 * @since TBD
+	 *
+	 * @param array<string,string> $columns The columns for the Attendees table.
+	 * @param int                  $event_id The event ID.
+	 *
+	 * @return array<string,string> The filtered columns for the Attendees table.
+	 */
+	public function add_attendee_seat_column( array $columns, int $event_id ): array {
+		$event_layout_id = get_post_meta( $event_id, Meta::META_KEY_LAYOUT_ID, true );
+		
+		if ( empty( $event_layout_id ) && ! tribe( Attendees_Page::class )->is_on_page() ) {
```

Comment:
This check on the current request being on the `Attendee_Page` is very late.
If we're not on the Attendee Page, by the time this bailed out, a number of queries and controller routing did happen.

I would move this check on this being the Attendee Page or not in the Controller and, there, hook this callback only if required.
Did you try that? Did it not work?